spec.md

```diff
@@ -134,41 +134,59 @@ the paths to label objects which can be found underneath the group:
 
 Unlisted groups MAY be labels.
 
-### "label""
+### "image-label" metadata
 
+Groups containing the `image-label` dictionary represent an image segmentation
+in which each unique pixel value represents a separate segmented object.
+`image-label` groups MUST also contain `multiscales` metadata and the two
+"datasets" series MUST have the same number of entries.
 
+The `colors` key defines a list of JSON objects describing the unique label
+values. Each entry in the list MUST contain the key "label-value" with the
+pixel value for that label. Additionally, the "rgba" key MAY be present, the
+value for which is an RGBA unsigned-int 4-tuple: `[uint8, uint8, uint8, uint8]`
+In the case that the same `label-value` is present multiple times in the list,
```

Comment:
Suggestion: simply say that a label value should not appear multiple times. Allows consumers to ignore a fiddly detail while suppliers probably weren't going to duplicate anyway.

Reply:
Agreed - easier to say 'don't do it'.

Reply:
I'm all for saying "don't do it", but I less clear about why it's not better to specify how to interpret data if someone does so. The options I can think of assuming one gets such data is:
 - treat first or last as definitive (requires parsing same amount of parsing)
 - throw an error (requires parsing)
 - leave it undefined

Reply:
Then we have to test the error handling or duplicate-key handling of the various consumers of the spec.
If it's not defined then it's up to each consumer of the zarr to handle it as is easiest for them.

Reply:
And indeed to ignore the issue altogether unless they're a validating parser because then such data would simply not be Zarr. Key duplication is but one of various specific quirks that could come up and it'd be great not to have to enumerate them, instead keeping the spec a clear, simple statement of the form of valid data.

Reply:
Agreed with tightening up the spec as suggested above. From past experience, knowingly allowing an ambiguous situation without defining the expectation usually comes back to bite us at some point. I would vote for either:

- using `must not` in the spec i.e. the consumers should error
- using `should not` in the spec  and suggest how consumers should be interpret such a duplication.

Obviously, coming back to https://github.com/ome/omero-ms-zarr/issues/62, it is also possible to enforce the non-duplication of label values via the structure of the colors metadata although as discussed there might be a trade-off with extensibility and/or usability.

Reply:
Yes - if we want the color values to be unique and the spec to enforce that, then use them as keys:
```
"label-metadata": {
  "1": {
    "rgba": [255, 255, 255, 0]
  },
  "4": {
    "rgba": [0, 255, 255, 128]
  },
}
```
Since the job of the 'colors' metadata is to map values to colors, a dictionary seems like the right choice.
Only use a list if the order is important.

Reply:
If JSON allowed non-string keys a dict would be ideal, but it doesn't so it just enables different errors to be made, e.g. it's quite easy to inadvertently mess up your int-to-string conversion in some languages and end up with things like `17`, `17.0`, `0017`, `1.7e1`, `0x11`

Reply:
OK, I'm happy with the current proposed spec and handling of duplicates with "the last value wins".

Reply:
I've just made a suggestion to change it to ``All `label-value`s must be unique.``